### PR TITLE
Feature/set up associations

### PR DESCRIPTION
**Pull request by MahdiAghaali**
In this branch I did the following activities and changes:

- Create the following classes
  - [Book](https://github.com/MahdiAghaali/school-library/commit/c8fd6c4465509313b0cd1af73b3dab8bda075b58)
  - [rental](https://github.com/MahdiAghaali/school-library/commit/7e9d6c0e1d108bc23c431185cac82fc7fa9381ac)
  - Classroom
- [Add method for assiging an student to a classroom](https://github.com/MahdiAghaali/school-library/commit/5434ed377518f2bfe81a2a48412f1373239efc64)
- [Create app.rb to run and test code](https://github.com/MahdiAghaali/school-library/commit/11753412231380d9b6ed5c978837d2ab45c38402)

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -9,4 +9,11 @@ def initialize(*person, classroom)
   def play_hooky
     '¯\(ツ)/¯'
   end
+
+  def student_class(param_class)
```
Comment (emmiiorji) [inline student.rb:13]: Kindly add `=` as a suffix to the name of the class to indicate that it performs an assignment. `student_class=`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+class Book
+  attr_accessor :title, :author, :rental_history
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental_history = []
+  end
+end
```
Comment (emmiiorji) [inline book.rb:11]: I think that you have forgotten to include a method to add a rental to its list of rentals. Kindly include it. For e.g
```
def add_rental(person, date)
    Rental.new(date, self, person)
end

```

File: person.rb [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
+  attr_accessor :name, :age, :rental_history
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = rand(1...1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental_history = []
   end
```
Comment (emmiiorji) [inline person.rb:18]: Similar to the above, kindly include a method to add a rental to a person's list of rentals.

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @MahdiAghaali, 
# Status: Changes Required ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Congrats

## Good Highlights✔️

- Linter checks are passing
- Good commit messages
- Good use of Gitflow
- Repo is professional


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MahdiAghaali** [comment]
@emmiiorji Thanks for feedback

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+require './classroom'
+
+puts '---------Test decorations--------'
+
+# puts person = Person.new(22, 'maximilianus')
+# puts person.correct_name
+# puts capitalized_person = CapitalizeDecorator.new(person)
+# puts capitalized_person.correct_name
+# puts capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
+# puts capitalized_trimmed_person.correct_name
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply app.rb]: Well resolved 👍🏽

File: student.rb (hunk 1 above)
Comment (Whoistolu) [reply student.rb]: Well resolved 👍🏽

File: book.rb (hunk 2 above)
Comment (Whoistolu) [reply book.rb]: Well resolved 👍🏽

File: person.rb (hunk 3 above)
Comment (Whoistolu) [reply person.rb:18]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @MahdiAghaali 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**